suite/runsuite.cmake

```diff
@@ -314,33 +318,38 @@ if (UNIX AND ARCH_IS_X86)
   set(ENV{CXXFLAGS} "")
   set(prev_run_tests ${run_tests})
   set(run_tests OFF) # build tests but don't run them
-  testbuild_ex("arm-debug-internal-32" OFF "
-    DEBUG:BOOL=ON
-    INTERNAL:BOOL=ON
-    BUILD_TESTS:BOOL=ON
-    CMAKE_TOOLCHAIN_FILE:PATH=${CTEST_SOURCE_DIRECTORY}/make/toolchain-arm32.cmake
-    " OFF OFF "")
-  testbuild_ex("arm-release-external-32" OFF "
-    DEBUG:BOOL=OFF
-    INTERNAL:BOOL=OFF
-    CMAKE_TOOLCHAIN_FILE:PATH=${CTEST_SOURCE_DIRECTORY}/make/toolchain-arm32.cmake
-    " OFF OFF "")
-  testbuild_ex("arm-debug-internal-64" ON "
-    DEBUG:BOOL=ON
-    INTERNAL:BOOL=ON
-    BUILD_TESTS:BOOL=ON
-    CMAKE_TOOLCHAIN_FILE:PATH=${CTEST_SOURCE_DIRECTORY}/make/toolchain-arm64.cmake
-    " OFF OFF "")
-  testbuild_ex("arm-release-external-64" ON "
-    DEBUG:BOOL=OFF
-    INTERNAL:BOOL=OFF
-    CMAKE_TOOLCHAIN_FILE:PATH=${CTEST_SOURCE_DIRECTORY}/make/toolchain-arm64.cmake
-    " OFF OFF "")
+  if (NOT cross_android_only)
+      testbuild_ex("arm-debug-internal-32" OFF "
+        DEBUG:BOOL=ON
+        INTERNAL:BOOL=ON
+        BUILD_TESTS:BOOL=ON
+        CMAKE_TOOLCHAIN_FILE:PATH=${CTEST_SOURCE_DIRECTORY}/make/toolchain-arm32.cmake
+        " OFF OFF "")
+      testbuild_ex("arm-release-external-32" OFF "
+        DEBUG:BOOL=OFF
+        INTERNAL:BOOL=OFF
+        CMAKE_TOOLCHAIN_FILE:PATH=${CTEST_SOURCE_DIRECTORY}/make/toolchain-arm32.cmake
+        " OFF OFF "")
+      testbuild_ex("arm-debug-internal-64" ON "
+        DEBUG:BOOL=ON
+        INTERNAL:BOOL=ON
+        BUILD_TESTS:BOOL=ON
+        CMAKE_TOOLCHAIN_FILE:PATH=${CTEST_SOURCE_DIRECTORY}/make/toolchain-arm64.cmake
+        " OFF OFF "")
+      testbuild_ex("arm-release-external-64" ON "
+        DEBUG:BOOL=OFF
+        INTERNAL:BOOL=OFF
+        CMAKE_TOOLCHAIN_FILE:PATH=${CTEST_SOURCE_DIRECTORY}/make/toolchain-arm64.cmake
+        " OFF OFF "")
+  endif (NOT cross_android_only)
   set(run_tests ${prev_run_tests})
 
   # Android cross-compilation and running of tests using "adb shell"
-  # FIXME i#2207: once we have Android cross-compilation working on Travis, remove this:
-  set(optional_cross_compile ON)
+  # For Travis cross_android_only builds, we want to fail on config failures.
+  # For user suite runs, we want to just skip if there's no cross setup.
+  if (NOT cross_android_only)
+      set(optional_cross_compile ON)
```

Comment:
I think there's a bug here: optional_cross_compile is never reset to OFF after it's set to ON for NOT cross_only above.